bgpd/bgp_vty.c

```diff
@@ -207,6 +197,40 @@ argv_find_and_parse_safi(struct cmd_token **argv, int argc, int *index, safi_t *
   return ret;
 }
 
+int
+bgp_vty_find_and_parse_afi_safi_vrf (struct vty *vty, struct cmd_token **argv, int argc, int idx,
+                                     afi_t *afi, safi_t *safi, vrf_id_t *vrf)
+{
+  char *vrf_name = NULL;
+
+  if (argv_find (argv, argc, "ip", &idx))
+      *afi = AFI_IP;
+
+  if (argv_find (argv, argc, "view", &idx) || argv_find (argv, argc, "vrf", &idx))
+    {
+      vrf_name = argv[idx + 1]->arg;
+      idx += 2;
+    }
+
+  if (argv_find_and_parse_afi (argv, argc, &idx, afi))
+    argv_find_and_parse_safi (argv, argc, &idx, safi);
+
+  if (vrf_name)
+    {
+      if (strmatch(vrf_name, "all"))
+       *vrf = VRF_ALL;
+      else
+       *vrf = vrf_name_to_id (vrf_name);
+    }
+
+  if (*vrf == VRF_UNKNOWN)
+    {
```

Comment:
I do not believe this is the right thing to do.  Only the calling function knows what the right defaults are for the afi/safi/vrf due to we assume different defaults based upon the command.  This approach would break the ability for the calling function to do the right thing.  As I stated I will update the Pull Request with better documentation for this funciton.